lib/presets.js

```diff
@@ -305,22 +305,55 @@ module.exports = {
      * });
      */
     cargo: (options, cwd, version, callback) => {
-      const cargoFiles = path.join(cwd, 'Cargo.*');
+      const cargoToml = path.join(cwd, 'Cargo.toml');
+      const cargoLock = path.join(cwd, 'Cargo.lock');
+
       const cleanedVersion = semver.clean(version);
 
       if (!cleanedVersion) {
         return callback(new Error(`Invalid version: ${version}`));
       }
 
-      replace({
-        files: cargoFiles,
-        from: /(version\s*=\s*)"(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)"/g,
-        to: '$1"' + cleanedVersion + '"'
-      }, (error) => {
-        if (error) {
-          return callback(error);
+      async.waterfall([
+        (done) => {
+          return fs.readFile(cargoToml, 'utf8', done);
+        },
+
+        (contents, done) => {
+          // Capture first `name = "..."` occurrence immediately after `[package]`
+          const matches = contents.match(/\[package\][^]+?name\s*=\s*("|')(.+)(\1)/m);
+          if (_.isNull(matches)) {
+            done(new Error(`Package name not found in ${cargoToml}`));
+          } else {
+            done(null, matches[2]);
+          }
+        },
+
+        (packageName, done) => {
+          if (fs.existsSync(cargoLock)) {
+            // Update first `version = "..."` occurrence immediately after `name = "${packageName}"`
+            const cargoLockRe = new RegExp(
+              `(name\\s*=\\s*(?:"|')${packageName}(?:"|')[^]+?version\\s*=\\s*)("|').*(?:\\2)`, 'm'
```

Comment:
nit: I think `(?:\\2)` can just be replaced by `\2`?
And I suspect you definitely want to replace the `.*` with `.*?`